mitodl/bootcamp-ecommerce
# fix anchor-tag related accessibility issues on dashboard

#### What are the relevant tickets?

closes #748

#### What's this PR do?

this fixes the accessibility issues on the application dashboard page relating to our use of `<a>` tags. We were using `<a>` tags with `onClick` handlers as a sort of hacky substitute for buttons.

We shouldn't do this. HTML defines a `<button>` tag which is designed for interactive elements which trigger some action without taking the user to another page. Screenreaders and browsers know what buttons are and how to handle them, while using `<a>` tags with `onclick` handlers is a sort of hacky and weird thing to do.

to switch over I:

- changed these tags over to be buttons
- added the `aria-expanded` attribute to the expand / collapse button to signal whether the content is expanded or not
- added a new `.borderless` class for buttons which makes them style more like an anchor tag. this should mean that there are no visual differences from what we were doing before.

#### How should this be manually tested?

do a full run-through of all the things on the application dashboard using the keyboard. you should be able to tab all around and do whatever you please (with the exception of triggering the resume upload, which is fixed by #912).
